packages/remix-dev/compiler.ts

```diff
@@ -337,8 +340,15 @@ async function createBrowserBuild(
     // All route entry points are virtual modules that will be loaded by the
     // browserEntryPointsPlugin. This allows us to tree-shake server-only code
     // that we don't want to run in the browser (i.e. action & loader).
-    entryPoints[id] =
-      path.resolve(config.appDirectory, config.routes[id].file) + "?browser";
+
+    // If any route entry point has *zero* browser safe exports (ie: only action/loader etc)
+    // then omit it entirely from the browser build
```

Comment:
I feel this comment would be clearer if inverted to match the if condition:

> Only include routes that export at least one browser-safe name (if not, it's a resource route and should only run on the server). 

Or something like that